libsql/src/local/connection.rs

```diff
@@ -38,14 +38,28 @@ impl Connection {
     pub(crate) fn connect(db: &Database) -> Result<Connection> {
         let mut raw = std::ptr::null_mut();
         let db_path = db.db_path.clone();
+        let flags: c_int = ffi::SQLITE_OPEN_FULLMUTEX | db.flags.bits();
+
         let err = unsafe {
+            if ffi::sqlite3_threadsafe() == 0 {
```

Comment:
I don't think this is actually safe, because accroding to the docs a user can still enable the threading model in such a way that it invalidates the assumptions the rust code has via `sqlite_confgi`

>  The return value of the sqlite3_threadsafe() function shows only the compile-time setting of thread safety, not any run-time changes to that setting made by sqlite3_config(). 

So this to me isn't strong enough to enforce 100% (esp with sqlite having global config) that the send/sync bounds we have in rust are safely implemented.